### PR TITLE
fix: re-render properly even in cases where getLayout returns a "pure" layout object

### PR DESCRIPTION
**Pull request by veinfors**
* Solves an issue with "ListboxPopoverContainer" not re-rendering on layout updates when used outside of sense-client. It was discovered when rendering through qmfe-embed where getLayout uses "pure layout" by default giving a new layout object every time.
* Removes onChange listeners properly which was not done before
* Adds prettier and pretty-quick to format commits automatically

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,6 +23,8 @@
     "@release-it-plugins/workspaces": "4.2.0",
     "@release-it/conventional-changelog": "5.1.1",
     "conventional-recommended-bump": "7.0.1",
+    "prettier": "^3.3.2",
+    "pretty-quick": "^4.0.0",
```
Comment (DanielS-Qlik) [inline package.json:27]: looks like husky or simple-git-hooks are needed for it to run as a commit hook: https://github.com/prettier/pretty-quick?tab=readme-ov-file#pre-commit-hook

File: .prettierrc [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "printWidth": 120,
+  "singleQuote": true,
+  "trailingComma": "es5",
```
Comment (DanielS-Qlik) [inline .prettierrc:4]:
```suggestion
  "trailingComma": "all",
```

File: packages/sn-filter-pane/src/components/ListboxGrid/use-handle-resize.ts [hunk 3]
```diff
@@ -146,10 +172,11 @@ export default function useHandleResize({
   };
 
   const resourceIds = resources.map((item) => item?.id).join(',');
+  const resourceLayouts = resources.map((item) => item?.layout); // make it react to new layout objects (fetched using "pure layout")
```
Comment (DanielS-Qlik) [inline packages/sn-filter-pane/src/components/ListboxGrid/use-handle-resize.ts:175]: isn't this creating a new `resourceLayouts` array on each render? (making the useCallback unnecessary)

Comment (veinfors) [reply packages/sn-filter-pane/src/components/ListboxGrid/use-handle-resize.ts:175]: oh, yes, you're right about that.. will adjust that

Comment (veinfors) [reply packages/sn-filter-pane/src/components/ListboxGrid/use-handle-resize.ts:175]: Done, think it should be fine now

**Review by DanielS-Qlik — APPROVED** [review]

**Pull request merged**